### PR TITLE
Filter Code Refactor & Fix Scroll on Focus Issues

### LINKED ISSUE #195: Internet Explorer scrolls on mouseover of last element
In Internet Explorer, when the mouse is passed over an element in the drop down list that is just barely visible, the element is scrolled in to view.

As far as I can tell from looking at the code, the reason this happens is because the input element is focused in this segment of code:

```
        .delegate('label', 'mouseenter.multiselect', function(){
            if( !$(this).hasClass('ui-state-disabled') ){
                self.labels.removeClass('ui-state-hover');
                $(this).addClass('ui-state-hover').find('input').focus();
            }
        })
```

commenting out that section causes the keyboard input to not work correctly so I assume it's a necessary evil; however, it's very annoying and bad for usability. 

If you know of any way to avoid this or can fix it, that would be great. Please don't hesitate to contact me if you want any more info on the bug.

Thanks!

### LINKED ISSUE #551: click to scroll issue
Hello,

I have a horizontal scrolling page and when click selectbox scrollbar going to left side

### PR DESCRIPTION
**Pull request by SteveTheTechie**
### multiselect.filter.js:

1.  Make filterRules a local variable.  (missed this before)
2.  Remove the unneeded multiselect() fxn call to get the widget instance--just use data() by itself.  (forgot to attempt this before)
3.  Save additional class names in variables that can be obfuscated away.
4.  Use ui-multiselect-grouplabel class name instead of finding the anchor tag directly.
5.  Decrease unneeded jQuery fxn calls w/ use of classList methods.

No feature changes.  All unit tests pass.

### multiselect.js

Fixes #195
Fixes #551

This fixes undesired scrolling issues caused by focusing an element that is only partially visible.
All unit tests pass.

### CODE REVIEW

**Comment by SteveTheTechie** [comment]
I intended this to be two different PRs but I forgot to create another branch in my repo to separate them.

File: src/jquery.multiselect.js [hunk 1]
```diff
@@ -1363,6 +1374,8 @@
       var speed = this.speed;
       var options = this.options;
       var effect = options.openEffect;
+		var scrollX = window.scrollX;
```
Comment (mlh758) [inline src/jquery.multiselect.js]: The spacing seems off here.

**Comment by mlh758** [comment]
That's alright, the changes are easy enough to review at once this time around. It looks like we're back to the tabs/spaces thing though on these changes.

**Comment by SteveTheTechie** [comment]
>  It looks like we're back to the tabs/spaces thing though on these changes.

Oops, sorry about that.  I forgot to check that.  Should be fixed now.

**Pull request merged**